### PR TITLE
Improve outbound listener conflict messages

### PR DESCRIPTION
**Pull request by nmittler**
This is needed to better support e2e testing of conflicts.

### CODE REVIEW

**Comment by nmittler** [comment]
@rshriram that's fine, but I'd ask that your PR include a better log message that indicates which service is having it's filterchain dropped.

**Comment by nmittler** [comment]
@rshriram @costinm PTAL  I've rebased on top of @rshriram's latest.

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 1]
```diff
@@ -348,6 +348,47 @@ func sortServicesByCreationTime(services []*model.Service) []*model.Service {
...
 
+func protocolName(p model.Protocol) string {
+	switch plugin.ModelProtocolToListenerProtocol(p) {
+	case plugin.ListenerProtocolHTTP:
+		return "HTTP"
+	case plugin.ListenerProtocolTCP:
+		return "TCP"
+	default:
+		return "UNKNOWN"
+	}
```
Comment (rshriram) [inline pilot/pkg/networking/core/v1alpha3/listener.go]: You shouldn't need this.. i think model.Protocol already has a string function that covers these.

Comment (nmittler) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: I don't see that function defined anywhere

**Pull request merged**